src/views/token-swap/components/amount-box/amount-box.view.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import useAmountBoxStyles from './amount-box.style'
+import { ReactComponent as AngleDown } from '../../../../images/icons/angle-down.svg'
+import FiatAmount from '../../../shared/fiat-amount/fiat-amount.view'
+import AmountInput from '../../../shared/amount-input/amount-input'
+import Dropdown from '../dropdown/dropdown.view'
+import { getTokenIcon } from '../../../../utils/tokens'
+import {
+  getTokenAmountInPreferredCurrency,
+  getFixedTokenAmount
+} from '../../../../utils/currencies'
+
+function AmountBox ({
+  account,
+  preferredCurrency,
+  fiatExchangeRates,
+  value,
+  amount,
+  onInputChange,
+  position,
+  accounts,
+  setToken,
+  setDropdown,
+  isDropdownActive
+}) {
+  const classes = useAmountBoxStyles()
+  const balance = getFixedTokenAmount(account?.balance, account?.token.decimals)
+  const setMax = () => {
+    onInputChange({
+      target: { value: balance }
+    })
+  }
+  const switchDropdown = () => {
+    const isActive = isDropdownActive ? '' : position
+    setDropdown(isActive)
+  }
+  const convertValue = getTokenAmountInPreferredCurrency(
+    value,
+    account?.token.USD,
+    preferredCurrency,
+    fiatExchangeRates
+  )
+  const Icon = getTokenIcon(account?.token.symbol)
+  const renderName = () => {
+    if (account) {
+      return (
+        <div className={classes.tokenName}>
+          <Icon className={classes.tokenIcon} /> {account.token.symbol}
+        </div>
+      )
+    }
+    return 'Select token'
+  }
+  const renderBalance = () => {
+    if (account) {
+      return (
+        <span className={classes.convertedText}>
```

Comment:
Maybe this outer element can be a <p> or a <div>. I would only use <span> if it's inline